src/Test/modellike.jl

```diff
@@ -471,3 +471,63 @@ function set_upper_bound_twice(model::MOI.ModelLike, T::Type)
         MOI.delete(model, ci)
     end
 end
+
+function delete_test(model::MOI.ModelLike)
+    x = MOI.add_variable(model)
+    cx = MOI.add_constraint(model, x, MOI.GreaterThan(0.0))
+    y = MOI.add_variables(model, 4)
+    cy = MOI.add_constraint(model, y, MOI.Nonpositives(4))
+    @test MOI.is_valid(model, x)
+    @test MOI.is_valid(model, y[1])
+    @test MOI.is_valid(model, y[2])
+    @test MOI.is_valid(model, y[3])
+    @test MOI.is_valid(model, y[4])
+    @test MOI.is_valid(model, cx)
+    @test MOI.is_valid(model, cy)
+    @test MOI.get(model, MOI.ConstraintFunction(), cx) == MOI.SingleVariable(x)
+    @test MOI.get(model, MOI.ConstraintSet(), cx) == MOI.GreaterThan(0.0)
+    @test MOI.get(model, MOI.ConstraintFunction(), cy) == MOI.VectorOfVariables(y)
+    @test MOI.get(model, MOI.ConstraintSet(), cy) == MOI.Nonpositives(4)
+    MOI.delete(model, y[3])
+    @test MOI.is_valid(model, x)
+    @test MOI.is_valid(model, y[1])
+    @test MOI.is_valid(model, y[2])
+    @test !MOI.is_valid(model, y[3])
+    @test MOI.is_valid(model, y[4])
+    @test MOI.is_valid(model, cx)
+    @test MOI.is_valid(model, cy)
+    @test MOI.get(model, MOI.ConstraintFunction(), cx) == MOI.SingleVariable(x)
+    @test MOI.get(model, MOI.ConstraintSet(), cx) == MOI.GreaterThan(0.0)
+    @test MOI.get(model, MOI.ConstraintFunction(), cy) == MOI.VectorOfVariables(y[[1, 2, 4]])
+    @test MOI.get(model, MOI.ConstraintSet(), cy) == MOI.Nonpositives(3)
+    MOI.delete(model, y[1])
+    @test MOI.is_valid(model, x)
+    @test !MOI.is_valid(model, y[1])
+    @test MOI.is_valid(model, y[2])
+    @test !MOI.is_valid(model, y[3])
+    @test MOI.is_valid(model, y[4])
+    @test MOI.is_valid(model, cx)
+    @test MOI.is_valid(model, cy)
+    @test MOI.get(model, MOI.ConstraintFunction(), cx) == MOI.SingleVariable(x)
+    @test MOI.get(model, MOI.ConstraintSet(), cx) == MOI.GreaterThan(0.0)
+    @test MOI.get(model, MOI.ConstraintFunction(), cy) == MOI.VectorOfVariables(y[[2, 4]])
+    @test MOI.get(model, MOI.ConstraintSet(), cy) == MOI.Nonpositives(2)
+    MOI.delete(model, x)
+    @test !MOI.is_valid(model, x)
+    @test !MOI.is_valid(model, y[1])
+    @test MOI.is_valid(model, y[2])
+    @test !MOI.is_valid(model, y[3])
+    @test MOI.is_valid(model, y[4])
+    @test !MOI.is_valid(model, cx)
```

Comment:
Perhaps also test that the deleted constraints don't appear in `ListOfConstraintIndices` and `ListOfConstraints`?